src/main/java/org/apache/sysds/runtime/matrix/data/LibMatrixDatagen.java

```diff
@@ -590,13 +615,22 @@ private static void genDense(MatrixBlock out, int clen, int blockrows, int block
 	}
 	
 	private static void genFullyDense(DenseBlock c, int blockrows, int blockcols, int rowoffset, int coloffset, 
-		double min, double range, PRNGenerator valuePRNG)
+		double min, double range, IPRNGenerator valuePRNG, long[] ctr)
 	{
-		for(int i = rowoffset; i < rowoffset+blockrows; i++) {
+		Iterator<Double> rngStream;
+		if (valuePRNG instanceof PRNGenerator) {
+			rngStream = Stream.generate(() -> min + (range * ((PRNGenerator) valuePRNG).nextDouble())).iterator();
+		} else if (valuePRNG instanceof CounterBasedPRNGenerator) {
+			rngStream = Arrays.stream(((CounterBasedPRNGenerator)valuePRNG).getDoubles(ctr, blockrows * blockcols)).map(i -> min + (range * i)).iterator();
```

Comment:
yup the stream looks nice, however, since it modifies old code, we need to be careful that the new code returns the same values as the previous.